src/main/java/org/opentripplanner/middleware/trip_monitor/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,176 @@
+        this.trip = trip;
+        this.injectedOtpResponse = injectedOtpResponse;
+    }
+
+    @Override
+    public void run() {
+        // Check if the trip check should be skipped (based on time, day of week, etc.)
+        if (shouldSkipMonitoredTripCheck(trip)) {
+            LOG.debug("Skipping check for trip: {}", trip.id);
+            return;
+        }
+        // Make a request to OTP with the monitored trip params or if this is in a test environment use the injected OTP
+        // response.
+        OtpDispatcherResponse otpDispatcherResponse = injectedOtpResponse != null
+            ? injectedOtpResponse
+            : OtpDispatcher.sendOtpPlanRequest(trip.queryParams);
+
+        if (otpDispatcherResponse.statusCode >= 400) {
+            // TODO: report bugsnag
+            LOG.error("Could not reach OTP server. status={}", otpDispatcherResponse.statusCode);
+            return;
+        }
+        Response otpResponse = otpDispatcherResponse.getResponse();
+        // TODO: Should null tripRequestId be fixed?
+        // TripSummary tripSummary = new TripSummary(otpDispatcherResponse.response.plan, otpDispatcherResponse.response.error, null);
+        // TODO: Find the specific itinerary to compare against. For now, just choose the first itin.
+        Itinerary itinerary = otpResponse.plan.itineraries.get(0);
+        // BEGIN CHECKS
+        // Check for notifications related to service alerts.
+        checkTripForNewAlerts(trip, itinerary);
+        // Check for notifications related to delays
+        checkTripForDelays(trip, itinerary);
+        // Check for notifications related to itinerary changes
+        checkTripForItineraryChanges(trip, itinerary);
+        // Add latest OTP response.
+        trip.addResponse(otpResponse);
+        Persistence.monitoredTrips.replace(trip.id, trip);
+        // Send notification to user.
+        sendNotification();
+    }
+
+    private void checkTripForItineraryChanges(MonitoredTrip trip, Itinerary itinerary) {
+        // TODO
+    }
+
+    private void checkTripForDelays(MonitoredTrip trip, Itinerary itinerary) {
+        // TODO
+    }
+
+
+    private void sendNotification() {
+        if (notifications.size() == 0) {
+            // FIXME: Change log level
+            LOG.info("No notifications queued up. Skipping notify.");
+            return;
+        }
+        OtpUser otpUser = Persistence.otpUsers.getById(trip.userId);
+        if (otpUser == null) {
+            LOG.error("Cannot find user for id {}", trip.userId);
+            // TODO: Bugsnag / delete monitored trip?
+            return;
+        }
+        String name = trip.tripName != null ? trip.tripName : "Trip for " + otpUser.email;
+        String subject = name + " Notification";
+        StringBuilder body = new StringBuilder();
+        for (TripMonitorNotification notification : notifications) {
+            body.append(notification.body);
+        }
```

Comment:
Ideally we'll have different text that gets sent out via sms/email, but I guess this is fine for now.